core/io/file_access_zip.cpp

```diff
@@ -147,8 +147,11 @@ unzFile ZipArchive::get_file_handle(String p_file) const {
 	return pkg;
 }
 
-bool ZipArchive::try_open_pack(const String &p_path, bool p_replace_files) {
+bool ZipArchive::try_open_pack(const String &p_path, bool p_replace_files, size_t p_offset = 0) {
 	//printf("opening zip pack %ls, %i, %i\n", p_name.c_str(), p_name.extension().nocasecmp_to("zip"), p_name.extension().nocasecmp_to("pcz"));
+	// load with offset feature only supported for PCK files
+	ERR_FAIL_COND_V_MSG(p_offset != 0, false, "Load with offset feature not supported for Zip files.");
```

Comment:
Loading Zip files with offset is not supported. Hence we will log an error message in debugger if the user attempts to load a Zip file with offset.

The trade-off for the current error handling is that if user attempts to load an invalid PCK file with offset, this error message will also be triggered and may appear out of context to the user.

Reply:
You could change the error message to make it more generic:

```text
"Invalid PCK data. Note that loading files with a non-zero offset isn't supported with ZIP archives."
```

Reply:
Thanks for the suggestion! I updated the error message accordingly.